dev/src/telemetry/enabled-trace-util.ts

```diff
@@ -42,6 +49,85 @@ export class EnabledTraceUtil implements TraceUtil {
     const libVersion = require('../../../package.json').version;
     const libName = require('../../../package.json').name;
     this.tracer = tracerProvider.getTracer(libName, libVersion);
+
+    this.settingsAttributes = {};
+    this.settingsAttributes['otel.scope.name'] = libName;
+    this.settingsAttributes['otel.scope.version'] = libVersion;
+
+    if (settings.projectId) {
+      this.settingsAttributes[`${ATTRIBUTE_SETTINGS_PREFIX}.project_id`] =
+        settings.projectId;
+    }
+
+    this.settingsAttributes[`${ATTRIBUTE_SETTINGS_PREFIX}.database_id`] =
+      settings.databaseId || DEFAULT_DATABASE_ID;
+
+    const host =
+      settings.servicePath ?? settings.host ?? 'firestore.googleapis.com';
+    const port = settings.port ?? FirestoreClient.port;
+    this.settingsAttributes[`${ATTRIBUTE_SETTINGS_PREFIX}.host`] =
+      `${host}:${port}`;
+
+    if (settings.preferRest !== undefined) {
+      this.settingsAttributes[`${ATTRIBUTE_SETTINGS_PREFIX}.prefer_REST`] =
+        settings.preferRest;
+    }
+
+    this.settingsAttributes[`${ATTRIBUTE_SETTINGS_PREFIX}.max_idle_channels`] =
+      settings.maxIdleChannels ?? DEFAULT_MAX_IDLE_CHANNELS;
+
+    const defaultRetrySettings = serviceConfig.retry_params.default;
```

Comment:
why there are retry and timeout configuration? how this is related to otel tracing?

Reply:
this code is adding trace attributes to the spans. The purpose is that if a user has misconfigured their SDK, their configurations will surface in the trace and may provide insights into the issues they're running into. This is similar to the [Java impl](https://github.com/googleapis/java-firestore/blob/1c7c12f915e76f25652adbad0864c5c7d810e268/google-cloud-firestore/src/main/java/com/google/cloud/firestore/telemetry/EnabledTraceUtil.java#L269). Although not all java configurations exist in the node.js SDK.